contracts/test/GrantRound.test.ts

```diff
@@ -0,0 +1,153 @@
+
+describe('GrantRound', function () {
+  let deployer: SignerWithAddress,
+    grantRoundOwner: SignerWithAddress,
+    donor: SignerWithAddress,
+    grantPayee: SignerWithAddress,
+    mpUser: SignerWithAddress;
+  let registry: GrantRegistry;
+  let roundContract: any;
+  let mockERC20: any;
+  const balances: string[] = ['100', '200'];
+  const startTime: number = Math.floor(new Date().getTime() / 1000); // time in seconds
+  const endTime: number = startTime + 86400; // One day
+
+  beforeEach(async () => {
+    const mockUrl: string = 'https://test.url';
+    const minContribution: number = 50;
+    [deployer, grantRoundOwner, donor, grantPayee, mpUser] = await ethers.getSigners();
+    mockERC20 = await deployMockContract(deployer, IERC20Artifact.abi);
+    // Seed funds for matching pool user
+    await mockERC20.mock.transfer.withArgs(mpUser.address, ethers.utils.parseEther(balances[0]));
+    // Seed funds for initial donor
+    await mockERC20.mock.transfer.withArgs(donor.address, ethers.utils.parseEther(balances[0]));
+
+    const grantRegistryArtifact: Artifact = await artifacts.readArtifact('GrantRegistry');
+    registry = <GrantRegistry>await deployContract(deployer, grantRegistryArtifact);
+
+    // Create an example grant
+    await registry.connect(deployer).createGrant(deployer.address, grantPayee.address, mockUrl);
+
+    const grantRoundArtifact: Artifact = await artifacts.readArtifact('GrantRound');
+    const roundContractFactory = new ethers.ContractFactory(
+      grantRoundArtifact.abi,
+      grantRoundArtifact.bytecode,
+      deployer
+    );
+
+    roundContract = await roundContractFactory.deploy(
+      grantRoundOwner.address,
+      registry.address,
+      mockERC20.address,
+      startTime,
+      endTime,
+      mockUrl,
+      minContribution
+    );
+  });
+
+  describe('Initialization', () => {
+    it('deploys properly', async function () {
+      expect(isAddress(roundContract.address), 'Failed to deploy GrantRegistry').to.be.true;
+    });
+  });
+
+  describe('acceptMatchingPool - Add funds to matching round', () => {
+    it('updates contract and user balances', async function () {
+      // set up mock calls
+      await mockERC20.mock.balanceOf.withArgs(mpUser.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.balanceOf.withArgs(roundContract.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.transferFrom.withArgs(mpUser.address, roundContract.address, balances[0]).returns(true);
+
+      await roundContract.connect(mpUser).acceptMatchingPool(balances[0]);
+      expect(await roundContract.connect(deployer).checkBalance(roundContract.address)).to.be.equal(
+        ethers.utils.parseEther(balances[0])
+      );
+    });
+  });
+
+  describe('donateToGrant - Add funds during active round', () => {
+    it('sends donation token to the grant payee with a given grant id', async function () {
+      const grantId: number = 0;
```

Comment:
Nit, but you don't need to specify the type here (and elsewhere where values are set). Typescript can infer it since you defined an initial number value
```suggestion
      const grantId = 0;
```